### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by shrushti2000**
1.  Added simple login ( Can ignore this commit, authentication will be implemented later on in detail)
2.  Added  add to cart and update quantity feature.
3.  Added cart summary calculations.
4. Implemented Remove from cart feature.

ezgif com-gif-maker (21)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,9 +2,12 @@ import { Route, Routes } from "react-router-dom";
 import "./App.css";
 import Header from "./components/Header/Header";
 import SubHeader from "./components/SubHeader/SubHeader";
+
 import TopBar from "./components/TopBar/TopBar";
+import CartPage from "./pages/CartPage/CartPage";
 import HomePage from "./pages/HomePage/HomePage";
 import ProductListingPage from "./pages/ProductListingPage/ProductListingPage";
+import Signin from "./pages/Signin/Signin";
```
Comment (junaidshaikh-js) [inline src/App.js:10]: You can create an index file in the pages folder and export all the components in the folder from this index file and then import from the pages folder only.

It makes the code cleaner. 

Refer to the Filter.jsx file in this PR 

https://github.com/junaidshaikh-js/autodecore-react/pull/9/files

File: src/pages/CartPage/CartPage.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import axios from 'axios'
+import React from 'react'
+import { useEffect } from 'react'
+import { useContext } from 'react'
```
Comment (junaidshaikh-js) [inline src/pages/CartPage/CartPage.js:4]: can be imported in single line

**Review by junaidshaikh-js — COMMENTED** [review]
Good Work! 

Suggested minor changes can be done. Everything else looks good. Can be merged.

**Review by AshwiniSetty — APPROVED** [review]

File: src/components/Header/Header.js [hunk 3]
```diff
@@ -1,9 +1,14 @@
 import React from 'react'
 import { Link } from 'react-router-dom'
+import { useContext } from 'react'
+import { StateContext } from '../../Context'
 import Searchbar from '../Searchbar/Searchbar'
 import './Header.css'
+import { useEffect } from 'react'
```
Comment (saifudinbadani) [inline src/components/Header/Header.js:7]: hooks can be imported in a single line

**Review by saifudinbadani — APPROVED** [review]
Code is readable and easy to understand 👍

File: src/App.js (hunk 1 above)
Comment (shrushti2000) [reply src/App.js:10]: Thanks @junaidshaikh-js for the suggestion and for sharing the reference. I am planning to refactor the folder and import structure soon.

**Comment by shrushti2000** [comment]
Thanks for the PR review @AshwiniSetty @saifudinbadani

**Pull request merged**